packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py

```diff
@@ -39,16 +38,15 @@
 from aea_test_autonomy.docker.base import skip_docker_tests
 
 from packages.valory.connections.ledger.connection import LedgerConnection
+from packages.valory.contracts.gnosis_safe.contract import (
+    PUBLIC_ID as GNOSIS_SAFE_PUBLIC_ID,
+)
 from packages.valory.protocols.contract_api.dialogues import ContractApiDialogue
 from packages.valory.protocols.contract_api.dialogues import (
     ContractApiDialogues as BaseContractApiDialogues,
 )
 from packages.valory.protocols.contract_api.message import ContractApiMessage
-
-
-GNOSIS_SAFE_PUBLIC_ID = PublicId.from_str("valory/gnosis_safe:0.1.0")
```

Comment:
The test works just fine like this. We can then fix it properly another time. I prefer it to stay in `ledger` connection as this actually tests that.

Reply:
So my problem with this being in the ledger connection package is, it still depends on the contract which not one of the dependency for the ledger connection. Another thing is we need to import and patch the `GnosisSafeContract` class for so many of the tests to work, which is not possible unless we define the contract as a dependency for the connection.

Take a look here you'll understand  https://github.com/valory-xyz/open-autonomy/pull/1329/files#diff-693e5a8d749782dd6ee62d30aad262db4c0a79c255704e32a24c588f675980fcL310-R332

Reply:
Ok